### PR TITLE
Add sub-section on renaming packages

### PR DESCRIPTION
**Pull request by mnquintana**
Adapted from https://github.com/atom/atom/blob/master/docs/apm-rest-api.md#renaming-a-package

/cc @atom/issue-triage

### CODE REVIEW

File: book/04-behind-atom/sections/07-maintaining-your-packages.asc [hunk 1]
```diff
@@ -35,3 +35,20 @@ apm unpublish package-name
 ```
 
 This will remove your package from the https://atom.io package registry. Anyone who has already downloaded a copy of your package will still have it and be able to use it, but it will no longer be available for installation by others.
+
+==== Rename Your Package
+
+You can rename your package by publishing a new version with the name changed in your package's `package.json`. Requests made to the previous name will forward to the new name.
```
Comment (50Wliu) [inline book/04-behind-atom/sections/07-maintaining-your-packages.asc]: `will be forwarded`?

**Comment by mnquintana** [comment]
I feel like this might still be missing one step - unpublishing the package under the old name? Here's one I just renamed from [carto-atom](https://atom.io/packages/carto-atom) to [language-carto](https://atom.io/packages/language-carto). carto-atom doesn't redirect to language-carto, and still exists in the registry. Does the old package name also need to be unpublished for this to work?

/cc @thedaniel

**Comment by thedaniel** [comment]
No, a rename should Just Work and redirect

**Comment by mnquintana** [comment]
Hmmm, any idea why [carto-atom](https://atom.io/packages/carto-atom) isn't redirecting to [language-carto](https://atom.io/packages/language-carto)?

**Comment by thedaniel** [comment]
You need to send a boolean `"rename": true` with the API call or it will create a new package. The `--rename` flag on `apm publish` does this for you, cf. `apm help publish`

**Comment by mnquintana** [comment]
Derp. So is the only step actually just running `apm publish --rename new-package-name` and that's it? You don't need to push a new release first or rename anything manually?

**Comment by mnquintana** [comment]
Alright, updated this with the most accurate instructions for renaming a package.

**Comment by mnquintana** [comment]
How does this look now @atom/issue-triage?

**Pull request merged**